test/sync/src/commonMain/kotlin/io/realm/test/mongodb/util/AdminApi.kt

```diff
@@ -305,16 +305,27 @@ open class AdminApiImpl internal constructor(
     // messages are being sent through our own node command server instead of using Ktor.
     private suspend fun sendPatchRequest(url: String, requestBody: JsonObject) {
         val forwardUrl = "$COMMAND_SERVER_BASE_URL/forward-as-patch"
-        client.request<HttpResponse>(forwardUrl) {
-            method = Post
-            parameter("url", url)
-            contentType(ContentType.Application.Json)
-            body = requestBody
-        }.let {
-            if (!it.status.isSuccess()) {
-                throw IllegalStateException("PATCH request failed: $it")
+
+        // It is unclear exactly what is happening, but if we only send the request once
+        // it appears as the server accepts it, but is delayed deploying the changes,
+        // i.e. the change will appear correct in the UI, but later requests against
+        // the server will fail in a way that suggest the change wasn't applied after all.
+        // Sending these requests twice seems to fix most race conditions.
+        repeat(2) {
```

Comment:
Couldn't this be solved instead by waiting for the change to be applied? sending the request twice may introduce further flakiness in the future, as we don't know exactly the root cause of the delay ...

Reply:
That was actually what I originally did, but even waiting for 5-10 seconds didn't improve things + it makes the tests quite a lot slower. For some reason sending the request twice works better :/